group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1128,6 +1401,26 @@ private void throwIfMemberEpochIsInvalid(
         }
     }
 
+    /**
+     * Validates the member epoch provided in the heartbeat request.
+     *
+     * @param member                The share group member.
+     * @param receivedMemberEpoch   The member epoch.
+     *
+     * @throws FencedMemberEpochException if the provided epoch is ahead or behind the epoch known
+     *                                    by this coordinator.
+     */
+    private void throwIfShareGroupMemberEpochIsInvalid(
+        ShareGroupMember member,
+        int receivedMemberEpoch
+    ) {
+        if (receivedMemberEpoch > member.memberEpoch()) {
```

Comment:
My understanding is that the member epoch is never persisted. Is it correct? If it is, it means that after a fail over of the group coordinator, all the members will be fenced because the member epoch here will be back to zero. Is it the case?

Reply:
Yes your understanding is right. The members will be fenced and will require to re-join with initial epoch. Do you recommend we should change this behaviour?